RecoTracker/MkFitCore/interface/HitStructures.h

```diff
@@ -0,0 +1,874 @@
+    void setOriginIndex(int oi) { m_origin_index = oi; }
+
+    void resetOverlaps() { m_overlap_hits.reset(); }
+    void considerHitForOverlap(int hit_idx, int module_id, float chi2) {
+      m_overlap_hits.consider_hit_for_overlap(hit_idx, module_id, chi2);
+    }
+    HitMatch* findOverlap(int hit_idx, int module_id) { return m_overlap_hits.find_overlap(hit_idx, module_id); }
+
+    // Inlines after definition of CombCandidate
+
+    HitOnTrack getLastHitOnTrack() const;
+    int getLastHitIdx() const;
+    int getLastHitLyr() const;
+
+    // For additional filter
+    int getLastFoundPixelHitLyr() const;
+    int getLastFoundHitLyr() const;
+    int nUniqueLayers() const;
+
+    int nLayersByTypeEncoded(const TrackerInfo& trk_inf) const;
+    int nHitsByTypeEncoded(const TrackerInfo& trk_inf) const;
+
+    int nPixelDecoded(const int& encoded) const { return encoded % 100; }
+    int nStereoDecoded(const int& encoded) const { return (encoded / 100) % 100; }
+    int nMonoDecoded(const int& encoded) const { return (encoded / 10000) % 100; }
+    int nMatchedDecoded(const int& encoded) const { return encoded / 1000000; }
+    int nTotMatchDecoded(const int& encoded) const {
+      return encoded % 100 + (encoded / 100) % 100 + (encoded / 10000) % 100 - encoded / 1000000;
+    }
+
+    void addHitIdx(int hitIdx, int hitLyr, float chi2);
+
+    HoTNode& refLastHoTNode();              // for filling up overlap info
+    const HoTNode& refLastHoTNode() const;  // for dump traversal
+
+    void incOverlapCount() { ++nOverlapHits_; }
+
+    Track exportTrack(bool remove_missing_hits = false) const;
+
+    void resetShortTrack() {
+      score_ = getScoreWorstPossible();
+      m_comb_candidate = nullptr;
+    }
+
+  protected:
+    CombCandidate* m_comb_candidate = nullptr;
+    HitMatchPair m_overlap_hits;
+
+    // using from TrackBase:
+    // short int lastHitIdx_
+    // short int nFoundHits_
+    short int nMissingHits_ = 0;
+    short int nOverlapHits_ = 0;
+
+    short int nInsideMinusOneHits_ = 0;
+    short int nTailMinusOneHits_ = 0;
+
+    short int m_origin_index = -1;  // index of origin candidate (used for overlaps in Standard)
+  };
+
+  inline bool sortByScoreTrackCand(const TrackCand& cand1, const TrackCand& cand2) {
+    return cand1.score() > cand2.score();
+  }
+
+  inline float getScoreCand(const TrackCand& cand1, bool penalizeTailMissHits = false, bool inFindCandidates = false) {
+    int nfoundhits = cand1.nFoundHits();
+    int noverlaphits = cand1.nOverlapHits();
+    int nmisshits = cand1.nInsideMinusOneHits();
+    int ntailmisshits = penalizeTailMissHits ? cand1.nTailMinusOneHits() : 0;
+    float pt = cand1.pT();
+    float chi2 = cand1.chi2();
+    // Do not allow for chi2<0 in score calculation
+    if (chi2 < 0)
+      chi2 = 0.f;
+    return getScoreCalc(nfoundhits, ntailmisshits, noverlaphits, nmisshits, chi2, pt, inFindCandidates);
+  }
+
+  // This inheritance sucks but not doing it will require more changes.
```

Comment:
please remove or clarify this comment

Reply:
Addressed in Round 1 follow-up PR trackreco/cmssw#75 commit: https://github.com/trackreco/cmssw/pull/75/commits/7fdc35d9228c436e832f8e9f37a9b7350e6b2c83
(the class inheritance was changed in response to another comment)